src/main/java/frc/robot/subsystems/pivot/PivotIOTalon.java

```diff
@@ -0,0 +1,216 @@
+    private final NeutralOut neutralControl = new NeutralOut().withUpdateFreqHz(0.0);
+
+    public PivotIOTalon() {
+        leaderPivotMotor = new TalonFX(PivotConstants.LEADER_PIVOT_MOTOR_ID);
+        followerPivotMotor = new TalonFX(PivotConstants.FOLLOWER_PIVOT_MOTOR_ID);
+        pivotEncoder = new CANcoder(PivotConstants.PIVOT_ENCODER_ID);
+        mmPositionRequest = new MotionMagicVoltage(0);
+        speakerAngleLookupValues = new SingleLinearInterpolator(PivotConstants.SPEAKER_PIVOT_POSITION);
+        passAngleLookupValues = new SingleLinearInterpolator(PivotConstants.PASS_PIVOT_POSITION);
+    
+        CANcoderConfiguration pivotEncoderConfig = new CANcoderConfiguration();
+        pivotEncoderConfig.MagnetSensor.MagnetOffset = -PivotConstants.ANGLE_ZERO;
+        pivotEncoderConfig.MagnetSensor.SensorDirection = PivotConstants.ENCODER_REVERSED;
+        pivotEncoder.getConfigurator().apply(pivotEncoderConfig, HardwareConstants.TIMEOUT_S);
+
+
+        TalonFXConfiguration pivotConfig = new TalonFXConfiguration();
+        pivotConfig.Slot0.kP = PivotConstants.PIVOT_P;
+        pivotConfig.Slot0.kI = PivotConstants.PIVOT_I;
+        pivotConfig.Slot0.kD = PivotConstants.PIVOT_D;
+        pivotConfig.Slot0.kG = PivotConstants.PIVOT_G;
+
+        pivotConfig.Slot0.GravityType = GravityTypeValue.Arm_Cosine;
+
+        pivotConfig.MotionMagic.MotionMagicAcceleration =
+            PivotConstants.MAX_VELOCITY_ROTATIONS_PER_SECOND;
+        pivotConfig.MotionMagic.MotionMagicCruiseVelocity =
+            PivotConstants.MAX_ACCELERATION_ROTATIONS_PER_SECOND_SQUARED;
+
+        pivotConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+        pivotConfig.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        pivotConfig.MotorOutput.DutyCycleNeutralDeadband = HardwareConstants.MIN_FALCON_DEADBAND;
+
+        pivotConfig.ClosedLoopGeneral.ContinuousWrap = true;
+
+        pivotConfig.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.RemoteCANcoder;
+        pivotConfig.Feedback.FeedbackRemoteSensorID = pivotEncoder.getDeviceID();
+
+        pivotConfig.SoftwareLimitSwitch.ForwardSoftLimitThreshold = PivotConstants.MAX_ANGLE;
+        pivotConfig.SoftwareLimitSwitch.ReverseSoftLimitThreshold = PivotConstants.MIN_ANGLE;
+        pivotConfig.SoftwareLimitSwitch.ForwardSoftLimitEnable = true;
+        pivotConfig.SoftwareLimitSwitch.ReverseSoftLimitEnable = true;
+        leaderPivotMotor.getConfigurator().apply(pivotConfig, HardwareConstants.TIMEOUT_S);
+        pivotConfig.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+        followerPivotMotor.getConfigurator().apply(pivotConfig, HardwareConstants.TIMEOUT_S);
+
+
+        leaderPivotMotor.getConfigurator().apply(pivotConfig, 1.0);
+        followerPivotMotor.getConfigurator().apply(pivotConfig, 1.0);
+        leaderPivotMotor.getConfigurator().apply(controllerConfig, 1.0);
```

Comment:
It looks like you guys aren't using controllerConfig. In addition, you are applying configs multiple times here, once up above and the extra stuff down here. I'd recommend removing the four down here for now.